From: ManoloFLTK <manolo.gouy@laposte.net>
Date: Tue, 28 Sep 2021 15:42:48 +0000
Subject: frame: Unlink from context when freeing

A wl_list_remove() statement matching the wl_list_insert() in
libdecor_decorate() should be added to function libdecor_frame_unref()
in file src/libdecor.c

The present code often crashes when several framed windows are created
and later destroyed.

Origin: upstream, 0.1.1, commit:51c6bf2aacc2359d87bf6c278dd1618f063596c3
---
 src/libdecor.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libdecor.c b/src/libdecor.c
index 59df8b7..36a46a9 100644
--- a/src/libdecor.c
+++ b/src/libdecor.c
@@ -600,6 +600,8 @@ libdecor_frame_unref(struct libdecor_frame *frame)
 		struct libdecor *context = frame_priv->context;
 		struct libdecor_plugin *plugin = context->plugin;
 
+		wl_list_remove(&frame->link);
+
 		if (frame_priv->xdg_toplevel)
 			xdg_toplevel_destroy(frame_priv->xdg_toplevel);
 		if (frame_priv->xdg_surface)
